packages/artifact/src/internal/upload-gzip.ts

```diff
@@ -9,15 +9,26 @@ const stat = promisify(fs.stat)
  * If any of these types of files are encountered then on-disk gzip creation will be skipped and the original file will be uploaded as-is
  */
 const gzipExemptFileExtensions = [
-  '.gzip',
-  '.zip',
-  '.tar.lz',
-  '.tar.gz',
-  '.tar.bz2',
-  '.tar.zst',
-  '.tar.zstd',
```

Comment:
Thanks for the great PR @hectorm! (really appreciate the tests) 🙇‍♂️ 

Requesting just a small change, could you add `.zstd`, currently it would be removed. Recently it was added as part of another PR https://github.com/actions/toolkit/pull/1184